hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/configuration/JobConfigurationConverter.java

```diff
@@ -0,0 +1,123 @@
+ * Copyright (C) 2013 - 2018, Logical Clocks AB and RISE SICS AB. All rights reserved
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the "Software"), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS  OR IMPLIED, INCLUDING
+ * BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL  THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR  OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.hops.hopsworks.common.jobs.configuration;
+
+import com.google.common.base.Strings;
+
+import java.io.StringReader;
+import java.io.StringWriter;
+import javax.persistence.AttributeConverter;
+import javax.persistence.Converter;
+import javax.ws.rs.core.MediaType;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.transform.stream.StreamSource;
+
+import io.hops.hopsworks.common.jobs.flink.FlinkJobConfiguration;
+import io.hops.hopsworks.common.jobs.jobhistory.JobType;
+import io.hops.hopsworks.common.jobs.spark.SparkJobConfiguration;
+import org.eclipse.persistence.jaxb.JAXBContextFactory;
+import org.eclipse.persistence.jaxb.MarshallerProperties;
+import org.json.JSONObject;
+
+@Converter
+public class JobConfigurationConverter implements AttributeConverter<JobConfiguration, String> {
+
+  @Override
+  public String convertToDatabaseColumn(JobConfiguration config) {
+    try {
+      JAXBContext jc = createJAXBContext(config.getJobType());
+      Marshaller marshaller = jc.createMarshaller();
+      marshaller.setProperty(MarshallerProperties.JSON_INCLUDE_ROOT, false);
+      marshaller.setProperty(MarshallerProperties.MEDIA_TYPE, MediaType.APPLICATION_JSON);
+      StringWriter sw = new StringWriter();
+      marshaller.marshal(config, sw);
+      return sw.toString();
+    } catch (JAXBException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public JobConfiguration convertToEntityAttribute(String jsonConfig) {
+    if (Strings.isNullOrEmpty(jsonConfig)) {
+      return null;
+    }
+    try {
+      JSONObject obj = new JSONObject(jsonConfig);
+      String jType = (String)obj.get("jobType");
+      JobType type = JobType.valueOf(jType);
```

Comment:
@robzor92 you can do this in one line as String `jType` and then JobType `type` is a bit confusing. For example `JobType.valueOf(JobType.valueOf(jType);`